core/coreapi/interface/options/dht.go

```diff
@@ -0,0 +1,62 @@
+package options
+
+type DhtProvideSettings struct {
+	Recursive bool
+}
+
+type DhtFindProvidersSettings struct {
+	NumProviders int
+}
+
+type DhtProvideOption func(*DhtProvideSettings) error
+type DhtFindProvidersOption func(*DhtFindProvidersSettings) error
+
+func DhtProvideOptions(opts ...DhtProvideOption) (*DhtProvideSettings, error) {
+	options := &DhtProvideSettings{
+		Recursive: false,
+	}
+
+	for _, opt := range opts {
+		err := opt(options)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return options, nil
+}
+
+func DhtFindProvidersOptions(opts ...DhtFindProvidersOption) (*DhtFindProvidersSettings, error) {
+	options := &DhtFindProvidersSettings{
+		NumProviders: 20,
+	}
+
+	for _, opt := range opts {
+		err := opt(options)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return options, nil
+}
+
+type dhtOpts struct{}
+
+var Dht dhtOpts
+
+// WithRecursive is an option for Dht.Provide which specifies whether to provide
+// the given path recursively
+func (dhtOpts) WithRecursive(recursive bool) DhtProvideOption {
```

Comment:
What's the rational behind the `With` prefix?

Reply:
Options before refractor used it, this didn't get updated with rebase. Will fix